src/script/components/Modals/WarningModal/WarningModalState.ts

```diff
@@ -0,0 +1,226 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import create from 'zustand';
+import {amplify} from 'amplify';
+import {WebAppEvents} from '@wireapp/webapp-events';
+import {Action, ModalContent, ModalItem, ModalOptions, ModalQueue, WarningModalType, Text} from './WarningModalTypes';
+import {getLogger} from 'Util/Logger';
+import {replaceLink, t} from 'Util/LocalizerUtil';
+import {noop} from 'Util/util';
+import {formatLocale} from 'Util/TimeUtil';
+import {ClientNotificationData} from '../../../notification/PreferenceNotificationRepository';
+import {Config} from '../../../Config';
+import {createRandomUuid} from 'Util/util';
+
+type WarningModalState = {
+  errorMessage: string | null;
+  queue: ModalQueue;
+  currentModalContent: ModalContent | null;
+  currentModalId: string | null;
+  existsInQueue: (modalItem: ModalItem) => boolean;
+  addToQueue: (modalItem: ModalItem) => void;
+  removeFirstItemInQueue: () => void;
+  replaceInQueue: (modalItem: ModalItem) => void;
+  updateCurrentModalId: (nextCurrentModalId: string | null) => void;
+  updateErrorMessage: (nextErrorMessage: string | null) => void;
+  updateCurrentModalContent: (nextCurrentModaContent: ModalContent | null) => void;
+};
+
+const defaultContent: ModalContent = {
+  checkboxLabel: '',
+  closeBtnTitle: '',
+  closeFn: noop,
+  currentType: null,
+  inputPlaceholder: '',
+  messageHtml: '',
+  messageText: '',
+  modalUie: '',
+  onBgClick: noop,
+  primaryAction: {} as Action,
+  secondaryAction: [],
+  titleText: '',
+};
+
+const logger = getLogger('WarningModalState');
+
+const useWarningModalState = create<WarningModalState>((set, get) => ({
+  addToQueue: (modalItem: ModalItem) =>
+    set(state => {
+      return {...state, queue: [...state.queue, modalItem]};
+    }),
+  currentModalContent: defaultContent,
+  currentModalId: null,
+  errorMessage: null,
+  existsInQueue: (modalItem: ModalItem): boolean =>
+    get().queue.findIndex(queueItem => queueItem.id === modalItem.id) !== -1,
+  queue: [],
+  removeFirstItemInQueue: () =>
+    set(state => {
+      return {...state, queue: [...state.queue.slice(1)]};
+    }),
+  replaceInQueue: (modalItem: ModalItem) =>
+    set(state => {
+      const updatedQueue = [...state.queue];
+      const index = updatedQueue.findIndex(queueItem => queueItem.id === modalItem.id);
+      updatedQueue[index] = modalItem;
```

Comment:
What do you think to use this (?) : 

```suggestion
      const updatedQueue = state.queue.map(queueItem => queueItem.id === modalItem.id ? modalItem : queueItem);
```